### PR TITLE
clang-tidy fixes

### PR DESCRIPTION
**Pull request by neheb**
Mostly readability fixes.

### CODE REVIEW

File: sys-utils/hwclock.c [hunk 1]
```diff
@@ -608,9 +608,8 @@ set_hardware_clock_exact(const struct hwclock_control *ctl,
 	}
 
 	newhwtime = sethwtime
-		    + (int)(time_diff(nowsystime, refsystime)
-			    - delay /* don't count this */
-			    + 0.5 /* for rounding */);
+		    + lround(time_diff(nowsystime, refsystime)
+			    - delay /* don't count this */);
```
Comment (karelzak) [inline sys-utils/hwclock.c:612]: Not sure if remove +0.5 is a good idea. Would it be possible that lround() returns time in the past?

File: disk-utils/cfdisk.c [hunk 2]
```diff
@@ -986,14 +986,15 @@ static size_t menuitem_get_line(struct cfdisk *cf, size_t idx)
...
-		size_t items = ui_cols / len;			/* items per line */
-
-		if (items == 0)
-			return 0;
-		return MENU_START_LINE + ((idx / items));
 	}
+
+	size_t len = MENU_H_ITEMWIDTH(m) + MENU_H_BETWEEN; /** item width */
+	size_t items = ui_cols / len;			/* items per line */
+
```
Comment (karelzak) [inline disk-utils/cfdisk.c:993]: This against our coding style, we define variables at beginning of the code block and this is the reason why there is else {}. Please, keep else {} there in this case.

File: disk-utils/fsck.c [hunk 3]
```diff
@@ -819,10 +819,10 @@ static struct fsck_instance *wait_one(int flags)
 		for (inst2 = instance_list; inst2; inst2 = inst2->next) {
 			if (inst2->flags & FLAG_DONE)
 				continue;
-			if (strcmp(inst2->type, "ext2") &&
+			if (strcmp(inst2->type, "ext2") != 0 &&
 			    strcmp(inst2->type, "ext3") &&
```
Comment (karelzak) [inline disk-utils/fsck.c:823]: Is it really really necessary to "fix" this issue? I guess we're smart enough to understand the code.

File: sys-utils/hwclock.c (hunk 1 above)
Comment (neheb) [reply sys-utils/hwclock.c:612]: maybe ceil() is better?

File: disk-utils/fsck.c (hunk 3 above)
Comment (neheb) [reply disk-utils/fsck.c:823]: well, != 0 is not necessary, but it is recommended. Hence why it's listed under bugprone.

File: sys-utils/hwclock.c (hunk 1 above)
Comment (karelzak) [reply sys-utils/hwclock.c:612]: ceil() sounds good.

File: disk-utils/fsck.c (hunk 3 above)
Comment (karelzak) [reply disk-utils/fsck.c:823]: OK :-)

**Pull request merged**